src/Formatter/FormatUserMentions.php

```diff
@@ -46,7 +47,9 @@ public function __invoke(Renderer $renderer, $context, string $xml)
         $post = $context;
 
         return Utils::replaceAttributes($xml, 'USERMENTION', function ($attributes) use ($post) {
-            $user = $post->mentionsUsers->find($attributes['id']);
+            $user = ($post instanceof Post && $post->exists)
+                ? $post->mentionsUsers->find($attributes['id'])
+                : User::find($attributes['id']);
```

Comment:
This can end up being very bad performance wise, each mention will produce a database query with those extensions, I would at least rather we check if the model has a `mentionsUsers` relationship instead, so that the extensions in question can still avoid the performance impact by mimicking the post model's `mentionsUsers` relationship.

`isset($post->getRelations()['mentionsUsers'])`

Also, would be better to use `$context` and drop the renaming to `$post`.

Reply:
Very good shout. Updated!